service/src/event/ordering_task.rs

```diff
@@ -1,158 +1,108 @@
 
-        let handle =
-            tokio::spawn(async move { Self::run_loop(pool, load_delivered, rx, rx_new).await });
+        let handle = tokio::spawn(async move {
+            Self::run_loop(pool, load_delivered, rx_delivered, rx_stream_update).await
+        });
 
         DeliverableTask {
             _handle: handle,
-            tx,
-            tx_new,
+            tx_delivered,
+            tx_stream_update,
         }
     }
 
     async fn run_loop(
         pool: SqlitePool,
         load_undelivered: bool,
-        mut rx: tokio::sync::mpsc::Receiver<DeliverableEvent>,
-        mut rx_new: tokio::sync::mpsc::Receiver<DeliveredEvent>,
+        mut rx_delivered: tokio::sync::mpsc::Receiver<CandidateEvent>,
+        mut rx_stream_update: tokio::sync::mpsc::Receiver<StreamCid>,
     ) {
         // before starting, make sure we've updated any events in the database we missed
         let mut state = OrderingState::new();
         if load_undelivered
             && state
-                .process_all_undelivered_events(&pool, MAX_ITERATIONS)
+                .process_all_undelivered_events(&pool)
                 .await
                 .map_err(Self::log_error)
                 .is_err()
         {
             return;
         }
 
-        let mut last_processed = std::time::Instant::now();
+        let mut interval =
+            tokio::time::interval(std::time::Duration::from_secs(CHECK_ALL_INTERVAL_SECONDS));
         loop {
-            let mut modified: Option<HashSet<InitCid>> = None;
-            let mut need_prev_buf = Vec::with_capacity(100);
-            let mut newly_added_buf = Vec::with_capacity(100);
+            let mut delivered_events = Vec::with_capacity(100);
+            let mut updated_streams = Vec::with_capacity(100);
 
             tokio::select! {
-                incoming = rx.recv_many(&mut need_prev_buf, 100) => {
-                    if incoming > 0 {
-                        modified = Some(state.add_incoming_batch(need_prev_buf));
-                    }
+                _new = rx_delivered.recv_many(&mut delivered_events, 100) => {
+                    debug!(?delivered_events, "incoming writes!");
                 }
-                new = rx_new.recv_many(&mut newly_added_buf, 100) => {
-                    if new > 0 {
-                        modified = Some(newly_added_buf.into_iter().map(|ev| ev.init_cid).collect::<HashSet<InitCid>>());
+                _new = rx_stream_update.recv_many(&mut updated_streams, 100) => {
+                    debug!(?updated_streams, "incoming updates!");
+                    for stream in updated_streams {
+                        state.add_stream(stream);
                     }
                 }
+                _ = interval.tick() => {
+                    // For this and the else branch, we allow the loop to complete and if state has anything to process, it will be processed.
```

Comment:
this comment confused me at first.  The loop doesn't complete in this case, we continue looping.  We just fall through this select! statement to the bottom of the loop body, then we loop back to the beginning.  Can we clarify this comment?

Reply:
removed most of the comments and took out the select!. hopefully it's clearer now, it's still a bit rough